### DISCUSSION TITLE
mirth csv to sql

### DISCUSSION THREAD

**Discussion by PrabhakaranTankaey** [nextgenhealthcare/connect#4951] [Q&A]
I am copy 100k records from csv and insert into sql.but it throws java heap size error.what is the effieciency way to insert morethan 100k record to sql server table with small interval.

**Comment by pacmano1** [comment]
The fastest way is to not use Mirth for large csv imports.  Instead use native command line database tools.

**Reply by PrabhakaranTankaey** [reply]
Hi,
native command line database tools mean can you please elaborate?
and mention what is major advantages of mirth and which format is good like hl7,csv,xls.

**Reply by StefanScholte** [reply]
for SQLServer there is  bcp utillity  https://www.sqlshack.com/techniques-to-bulk-copy-import-and-export-in-sql-server/

**Reply by pacmano1** [reply]
For simple data extracts where you do not need to modify the data during the extract bcp will be faster.   Mirth does not natively support Excel file (xls) exports.   If you need to export to hl7, then yes use Mirth because you need to transform data to a hl7 complaint message structure which is very different than a csv or xls file.

**Reply by PrabhakaranTankaey** [reply]
@pacmano1  is it possible to read multiple hl7 message at a time.using file reader

**Reply by pacmano1** [reply]
Yes.
